javaslang/src/main/java/javaslang/concurrent/Future.java

```diff
@@ -309,10 +309,45 @@
      * @throws NullPointerException if one of executorService of computation is null.
      */
     static <T> Future<T> of(ExecutorService executorService, CheckedSupplier<? extends T> computation) {
+        return Future.of(executorService, computation, 0);
+    }
+
+    /**
+     * Starts an asynchronous computation after delaying, backed by the {@link #DEFAULT_EXECUTOR_SERVICE}.
+     * The thread supplied by the {@code DEFAULT_EXECUTOR_SERVICE} will be blocked at least {@code delay} milliseconds.
+     * If the {@code delay} &lt;= 0, the {@code computation} will be executed immediately.
+     *
+     * @param computation A computation.
+     * @param delay    The time in milliseconds to wait before executing the given {@code computation}.
+     * @param <T>         Type of the computation result.
+     * @return A new Future instance.
+     * @throws NullPointerException if computation is null.
+     */
+    static <T> Future<T> of(CheckedSupplier<? extends T> computation, long delay) {
+        return Future.of(DEFAULT_EXECUTOR_SERVICE, computation, delay);
+    }
+
+    /**
+     * Starts an asynchronous computation after delaying, backed by the given {@link ExecutorService}.
+     * The thread supplied by the {@code ExecutorService} will be blocked at least {@code delay} milliseconds.
+     * If the {@code delay} &lt;= 0, the {@code computation} will be executed immediately.
+     *
+     * @param executorService An executor service.
+     * @param computation     A computation.
+     * @param delay        The time in milliseconds to wait before executing the given {@code computation}.
+     * @param <T>             Type of the computation result.
+     * @return A new Future instance.
+     * @throws NullPointerException if one of executorService of computation is null.
+     */
+    static <T> Future<T> of(ExecutorService executorService, CheckedSupplier<? extends T> computation, long delay) {
         Objects.requireNonNull(executorService, "executorService is null");
         Objects.requireNonNull(computation, "computation is null");
         final FutureImpl<T> future = new FutureImpl<>(executorService);
-        future.run(computation);
+        final CheckedSupplier<? extends T> task = (delay <= 0) ? computation : () -> {
+            Thread.sleep(delay);
```

Comment:
@hepin1989 which features of Javaslang do you use mainly?

Are you interested in a guest blog-post about how you use Javaslang? I.e. on the [Javaslang Blog](http://blog.javaslang.io)